docs/topic-guides/workflows.rst

```diff
@@ -0,0 +1,100 @@
+.. _workflows:
+
+Renku Workflows
+===============
+
+Together with :ref:`provenance of results <provenance>`, Renku workflows is
+a key feature of Renku to make code and processing pipelines reusable.
+When tracking executions with Renku, each execution creates a workflow step.
+These steps can be joined together into bigger workflows, making it easy to
+run them with varying inputs and on different execution backends.
+
+There are many uses for this. You could define and test a pipeline locally
+and then execute it on an HPC cluster on a big dataset once you're happy.
+Or you could run your workflow offer a range of different parameters to see
+how the results differ. Or you create a workflow that runs daily with a
+CRON job.
+
+All executions of workflows are also tracked in the provenance, without you
+having to re-type and memorize the commands used in each step.
+
+Working with Workflows
+----------------------
+
+Tracking execution of a command with :meth:`renku run <renku.ui.cli.run>`:
+
+.. code-block:: console
+
+    $ renku run --name run-analysis -- python run_analysis.py -i inputs -o outputs
+
+creates a workflow step called `run-analysis`. You can inspect the workflow
+with :meth:`renku workflow show <renku.ui.cli.workflow>`:
+
+.. code-block:: console
+
+    $ renku workflow show run-analysis
+    Id: /plans/76d73efb94964e9aac3635176ea57a36
+    Name: run-analysis
+    Creators: John Doe <example@renku.ch>
+    Command: python run_analysis.py -i inputs -o outputs
+    Success Codes:
+    Inputs:
+            - input-1:
+                    Default Value: run_analysis.py
+                    Position: 1
+            - i-2:
+                    Default Value: inputs
+                    Position: 2
+                    Prefix: -i
+    Outputs:
+            - o-3:
+                    Default Value: outputs
+                    Position: 3
+                    Prefix: -o
+
+You can execute it with :meth:`renku workflow execute <renku.ui.cli.workflow>`:
```

Comment:
I recommend demonstrating basic re-execution of the workflow before introducing modifying the workflow parameters.